### ISSUE TITLE
ctverif doesn't work for bam version 1.0.6

### ISSUE THREAD

**Issue by shaobo-he** [michael-emmi/ctverif#1]
```
ctverif version 1.0.0
SMACK program verifier version 1.9.0
Traceback (most recent call last):
	11: from /uusoc/exports/scratch/shaobo/gem/bin/bam:23:in `<main>'
	10: from /uusoc/exports/scratch/shaobo/gem/bin/bam:23:in `load'
	 9: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/exe/bam:4:in `<top (required)>'
	 8: from /uusoc/exports/scratch/shaobo/ruby-2.5.1/install/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 7: from /uusoc/exports/scratch/shaobo/ruby-2.5.1/install/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 6: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:148:in `<top (required)>'
	 5: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bam/prelude.rb:135:in `timed'
	 4: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:150:in `block in <top (required)>'
	 3: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:150:in `each'
	 2: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:150:in `block (2 levels) in <top (required)>'
	 1: from /uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bpl/passes/security/shadowing.rb:605:in `run!'
/uusoc/exports/scratch/shaobo/gem/gems/bam-bam-boogieman-1.0.6/lib/bpl/passes/security/shadowing.rb:14:in `parse_args': undefined method `split' for nil:NilClass (NoMethodError)
SMACK result is unknown.
```

I got the following message when using ctverif against bam version 1.0.6. When I switched to bam version 1.0.1.alpha4 everything worked fine.

**Comment by michael-emmi** [comment]
Please provide input code to reproduce.

**Comment by bjohannesmeyer** [comment]
I'm running into what looks like the same issue. 

I've created a [Docker image](https://hub.docker.com/r/bjohannesmeyer/ctverif_updated/)(*) that can reproduce this, with SMACK 1.9.0 and Bam 1.0.6. To use it (warning: image size ~1GB), run:

```
docker pull bjohannesmeyer/ctverif_updated
docker run -it bjohannesmeyer/ctverif_updated /bin/bash
```

This will launch a shell into the Docker container. Then make a directory for the example:
```
mkdir ~/tmp && cd ~/tmp
```

Then create ct-verif.h as well as a foo.c as follows:
```
#include "ct-verif.h"

void foo(int x) {
        int a;
        if(x == 3) a = 9*7;
}

void foo_wrapper(int x) {
        foo(x);
}
```

Then running `ctverif foo.c` yields:
```
ctverif version 1.0.0
SMACK program verifier version 1.9.0
/var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bpl/passes/security/shadowing.rb:14:in `parse_args': undefined method `split' for nil:NilClass (NoMethodError)
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bpl/passes/security/shadowing.rb:605:in `run!'
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:150:in `block (2 levels) in <top (required)>'
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:150:in `each'
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:150:in `block in <top (required)>'
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bam/prelude.rb:135:in `timed'
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/lib/bam.rb:148:in `<top (required)>'
        from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /usr/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
        from /var/lib/gems/2.3.0/gems/bam-bam-boogieman-1.0.6/exe/bam:4:in `<top (required)>'
        from /usr/local/bin/bam:23:in `load'
        from /usr/local/bin/bam:23:in `<main>'
SMACK result is unknown.
```
The output looks slightly different I think because I'm using Ruby 2.3.0 and it looks like @shaobo-he is using 2.5.0. 

(*)The image was produced from the following Dockerfile:
```
FROM ubuntu:16.04
RUN apt-get -y update
RUN apt-get -y install git npm nodejs-legacy wget
RUN npm install -g n && n lts

RUN apt-get -y install mono-complete
RUN apt-get -y install lsb-release sudo software-properties-common
RUN cd /root && git clone https://github.com/smackers/smack.git -b develop
RUN cd /root/smack/bin && ./build.sh
ENV BOOGIE="mono /root/boogie/Binaries/Boogie.exe"
ENV CORRAL="mono /root/corral/bin/Release/corral.exe"

RUN apt-get -y upgrade

RUN apt-get -y install ruby-dev
RUN gem install bam-bam-boogieman bundler

RUN npm i -g ctverif

RUN apt-get -y install vim 
```

**Comment by michael-emmi** [comment]
Just pushed fixes; please update to ctverif 1.0.1 and bam 1.0.7

**Comment by bjohannesmeyer** [comment]
That fixed it. Thank you very much for all of your help!
